jenkinsbuilder/indexdindexymltransform.py

```diff
@@ -0,0 +1,182 @@
+                    "notify-email": entry["notify-email"],
+                    "depends-on": entry["depends-on"]
+                })
+
+                current_index += 1
+
+        # Dump the indexyml data to specfied index.yml
+        with open(self._index_yml, "w") as index_yml_file: 
+            safe_dump(index_yml, index_yml_file, default_flow_style=False, indent=2)
+
+        # Add few lines of comment indicating the the file is generated.
+        commentline = """This file is generated by the CentOS Community Container Pipeline Service\n
+                      Please do not edit this file manually.\n\n
+                      """
+
+        with open(self._index_yml, "r") as original:
+            data = original.read()
+
+        with open(self._index_yml, "w") as modified:
+            modified.write(data)
+
+        return
+
+    def _genindexd(self):
+        """Generate index.d files given index.yml"""
+
+        index_yml = None
+        indexd_data = {}
+
+        # Open the indexyml file and read the yaml data
+        with open(self._index_yml, "r") as index_file:
+            index_yml = load(index_file)
+
+        # For every project in the indexyml file, collect the necessary data
+        for project in defaultdict(index_yml["Projects"]):
+            app_id = project["app-id"]
+
+            # If no desired-tag is specified, assume a default, otherwise take specified desired-tag
+            if "desired-tag" not in project:
+                desired_tag = "latest"
+
+            else:
+                desired_tag = project["desired-tag"]
+
+            # If there isnt already indexd data entry for a specific appid, add it.
+            # The appid will be used later on to set the filenames
+            if app_id not in indexd_data:
+                indexd_data[app_id] = []
+
+            # Append the collected and generated data to indexd data
+            indexd_data[app_id].append({
+                "job-id": project["job-id"],
+                "git-url": project["git-url"],
+                "git-path": project["git-path"],
+                "git-branch": project["git-branch"],
+                "target-file": project["target-file"],
+                "notify-email": project["notify-email"],
+                "desired-tag": desired_tag,
+                "depends-on": project["depends-on"]
+            })
+
+        # Iterate through the indexd data, reading
+        # key : appid, name of the index.d file (appid.yml)
+        # value : list of entries to go into that file
+        for key, value in indexd_data.items():
+
+            # Dump the yml value to the key.yml in index.d
+            if key is not None:
+                file_name = self._index_dotd + "/" + key + ".yml"
+
+                if os.path.exists(file_name):
+                    os.remove(file_name)
+                with open(file_name, "a+") as the_file:
+                    safe_dump(value, the_file, default_flow_style=False)
+
+        return
```

Comment:
@mohammedzee1000 

I don't see how is that gonna fix the problem. You are not expecting any output from ``_genindex`` anyways, so why bother to have the return statement. If there's a corner case somewhere else, fix that there.